### PR TITLE
chore(update-mocha): Update mocha to latest version

### PR DESCRIPTION
**Pull request by pdufour**
This fixes the security vulnerabilities we are seeing.

I.e.

```
yarn audit v1.21.1
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimatch                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mocha                                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mocha > glob > minimatch                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/118      
```

### CODE REVIEW

**Comment by hanqingchen15** [comment]
why are so many packages added to `yarn.lock` file? Are they all dependencies of the newer `mocha` version?

File: yarn.lock [hunk 1]
```diff
@@ -51,109 +168,568 @@ deep-eql@0.1.3:
...
-  integrity sha1-FhvlvetJZ3HrmzV0UFC2IrWu/Fg=
-  dependencies:
-    commander "2.3.0"
-    debug "2.2.0"
-    diff "1.4.0"
-    escape-string-regexp "1.0.2"
-    glob "3.2.11"
-    growl "1.9.2"
-    jade "0.26.3"
+mocha@^7.1.0:
```
Comment (pdufour) [inline yarn.lock]: @hanqingchen15

Comment (pdufour) [reply yarn.lock]: Lots of new packages here, sub-deps of Mocha

**Comment by pdufour** [comment]
i'll rebase. thx @Luke-Py-Walker @hanqingchen15 .

**Pull request merged**